Tilers/IfcTiler/ifcObjectGeom.py

```diff
@@ -0,0 +1,297 @@
+                    continue
+                indexListTemp = None
+                vertexListTemp = None
+                if(representation.RepresentationType == "Tessellation") :
+                    indexListTemp = itemGeom.CoordIndex
+                    vertexListTemp = itemGeom.Coordinates.CoordList
+                elif(representation.RepresentationType == "SweptSolid") :
+                    vertexListTemp,indexListTemp = self.extrudGeom(itemGeom)
+                if(vertexListTemp and indexListTemp) :
+                    for index in indexListTemp :
+                        indexList.append([index[0]+len(vertexList),index[1]+len(vertexList),index[2]+len(vertexList)])
+                    for vertex in vertexListTemp:
+                        vertexList.append(np.array([vertex[0],vertex[1],vertex[2]],dtype=np.float32))
+
+
+        if (len(indexList) == 0) :
+            return False
+
+        for j in range(len(vertexList)):
+            vertex = vertexList[j]
+            for i in range(len(listDirection)) :
+                vertex = np.dot(np.array(vertex),listDirection[i])
+                vertex = (vertex + listPosition[i])
+            if(self.convertToMeter):
+                vertex = vertex / 100
+            vertexList[j] = np.array([round(vertex[0],5),round(vertex[1],5),round(vertex[2],5)],dtype=np.float32)
+
+        triangles = list()
+        for index in indexList:
+            triangle = []
+            for i in range(0,3): 
+                # We store each position for each triangles, as GLTF expect
+                triangle.append(vertexList[index[i] - 1])
+            triangles.append(triangle)
+        
+        self.geom.triangles.append(triangles)
+
+        self.set_box()
+
+        return True
+    
+    def set_box(self):
+        """
+        Parameters
+        ----------
+        Returns
+        -------
+        """
+        bbox = self.geom.getBbox()
+        self.box = BoundingVolumeBox()
+        self.box.set_from_mins_maxs(np.append(bbox[0],bbox[1]))
+        
+        # Set centroid from Bbox center
+        self.centroid = np.array([(bbox[0][0] + bbox[1][0]) / 2.0,
+                         (bbox[0][1] + bbox[1][1]) / 2.0,
+                         (bbox[0][2] + bbox[0][2]) / 2.0])
+
+    def get_obj_id(self):
+        return super().get_id()
+    
+    def set_obj_id(self,id):
+        return super().set_id(id)
+
+class IfcObjectsGeom(ObjectsToTile):
+    """
+        A decorated list of ObjectsToTile type objects.
+    """
+    def __init__(self,objs=None):
+        super().__init__(objs)
+
+    @staticmethod
+    def computeCentroid(ifcSite,convertToMeter) :
+        elevation = ifcSite.RefElevation
+        placement = ifcSite.ObjectPlacement.RelativePlacement
+        location = placement.Location.Coordinates
+        transformer = Transformer.from_crs("EPSG:27562", "EPSG:3946")
+        if(convertToMeter) :
+            location = (location[0]/100,location[1]/100,location[2] / 100)
```

Comment:
Try to localize in a single place the treatment of the units (to avoid scattered repetitions AND being polluted by many units)...

Reply:
Opened [an issue for this](https://github.com/VCityTeam/py3dtiles-tilers/issues/4)